### PR TITLE
Travis 1.20 backports

### PR DESCRIPTION
**Pull request by raveit65**
back port of https://github.com/mate-desktop/caja/commit/91f1dc66b6f56e57734d3048501a6c4679331290 and https://github.com/mate-desktop/caja/commit/2d595142c4c3f15391d71bf25e48e5905804ba94

### CODE REVIEW

**Comment by raveit65** [comment]
Ohh, looks like 1.20 branch isn't at travis-CI

**Comment by raveit65** [comment]
Hmm, archlinux failed with
`Error response from daemon: manifest for base/archlinux:latest not found`

**Comment by yetist** [comment]
It seems that the https://travis-ci.org website is temporarily unable to download the archlinux image from https://hub.docker.com. After a while, try clicking on the **Restart job** on the website.

Travis-CI is valid on all branches, maybe you can try to delete the .travis.yml on this branch?

**Comment by raveit65** [comment]
Without .travis.yml on 1.20 branch we get this issue `Build config file is required`
https://travis-ci.org/mate-desktop/caja/requests

**Comment by yetist** [comment]
Maybe I can modify docker-build to stop executing and return 0 when download the docker image fails？

**Comment by raveit65** [comment]
In case if it fails only for one distro it is OK.
But what happens if travis can't download docker images for other distros?
Than we have  succesfull checks without a build :)
Let's wait a day, sometimes internet is bad.
Maybe tomorrow it's better.
Btw. we got same issue with marco and 1.20 branch.

**Comment by yetist** [comment]
@raveit65 

You are right, this branch also need the `.travis.yml` file. I created the `test-ci-active` branch with tag `v1.20.3` and made a commit, opened a PR, and the CI is not activated.

OK, let's wait a day, sometimes internet is bad.

**Comment by yetist** [comment]
At the end of this link https://docs.travis-ci.com/user/customizing-the-build/#building-specific-branches, I read this:
> Note that for historical reasons .travis.yml needs to be present on all active branches of your project.

Maybe this is the reason.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    - quilt
+    - shared-mime-info
+
+# Uncomment the following lines to get more compile warnings
+variables:
+  - CFLAGS="-Wall -Werror=format-security"
+
+after_scripts:
+  - make distcheck
+  - 'if [ $? -ne 0 ];then RED="\033[0;31m"; ENDC="\033[0m"; printf "${RED}!!! ERROR: Run make distcheck failed.${ENDC}\n"; fi'
```
Comment (yetist) [inline .travis.yml]: I think the 121 line can remove now, because the distcheck of all distros on the master branch are passed.

**Pull request merged**